### ISSUE TITLE
[mingw64] [ffmpeg build error.  openal not found]

### ISSUE THREAD

**Issue by jkchoi2** [jb-alvarado/media-autobuild_suite#1324]
``` bash
[1m┌ ffmpeg git (B[m ........................................... [[93mFiles missing(B[m]
[1m├(B[m Changing options to comply to nonfree...
[93mFFmpeg and related apps will depend on CUDA SDK to run!(B[m
[93mzLsN.txt(B[m
	Patch couldn't be applied with 'git am'. Continuing without patching.
[1m├(B[m Compiling [1mshared(B[m FFmpeg...
[1m├(B[m Running configure...
Likely error:
CFLAGS: -mthreads -mtune=generic -O2 -pipe
LDFLAGS: -pipe -static-libgcc -static-libstdc++ -L/local64/lib -L/mingw64/lib
../configure --disable-static --enable-shared --pkg-config-flags=--static --disable-autodetect --enable-amf --enable-bzlib --enable-cuda --enable-cuvid --enable-d3d11va --enable-dxva2 --enable-iconv --enable-lzma --enable-nvenc --enable-zlib --enable-sdl2 --disable-debug --enable-ffnvcodec --enable-nvdec --enable-libmp3lame --enable-libopus --enable-libvorbis --enable-libvpx --enable-libx264 --enable-libx265 --enable-libdav1d --enable-fontconfig --enable-libass --enable-libbluray --enable-libfreetype --enable-libmfx --enable-libmysofa --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libopenjpeg --enable-libsnappy --enable-libsoxr --enable-libspeex --enable-libtheora --enable-libtwolame --enable-libvidstab --enable-libvo-amrwbenc --enable-libwavpack --enable-libwebp --enable-libxml2 --enable-libzimg --enable-libshine --enable-gpl --enable-avisynth --enable-libxvid --enable-libaom --enable-libopenmpt --enable-version3 --enable-chromaprint --enable-decklink --enable-frei0r --enable-libbs2b --enable-libcaca --enable-libcdio --enable-libfdk-aac --enable-libflite --enable-libfribidi --enable-libgme --enable-libgsm --enable-libilbc --enable-libkvazaar --enable-libmodplug --enable-libnpp --enable-libopenh264 --enable-librtmp --enable-librubberband --enable-libssh --enable-libtesseract --enable-libxavs --enable-libzmq --enable-libzvbi --enable-openal --enable-opencl --enable-opengl --enable-libvmaf --enable-libcodec2 --enable-libsrt --enable-ladspa --enable-libsvthevc --enable-openssl --extra-cflags=-fopenmp --extra-libs=-lgomp --extra-cflags=-DLIBTWOLAME_STATIC --extra-libs=-lstdc++ --extra-cflags=-DLIBSSH_STATIC --extra-ldflags=-Wl,--allow-multiple-definition --extra-cflags=-DCACA_STATIC --extra-cflags=-DMODPLUG_STATIC --extra-cflags=-DCHROMAPRINT_NODLL --extra-libs=-lstdc++ --extra-cflags=-DZMQ_STATIC --extra-libs=-lpsapi --extra-cflags=-DLIBXML_STATIC --extra-libs=-liconv --disable-w32threads --extra-cflags=-DKVZ_STATIC_LIB --enable-nonfree --extra-cflags=-ID:/Cuda/include --extra-ldflags=-LD:/Cuda/lib/x64 --prefix=/local64/bin-video/ffmpegSHARED
ERROR: openal not found

If you think configure made a mistake, make sure you are using the latest
version from Git.  If the latest version fails, report the problem to the
ffmpeg-user@ffmpeg.org mailing list or IRC #ffmpeg on irc.freenode.net.
Include the log file "ffbuild/config.log" produced by configure as this will help
solve the problem.
[31mconfigure failed. Check D:/media-autobuild_suite/build/ffmpeg-git/build-shared-64bit/ab-suite.configure.log(B[m
[31mThis is required for other packages, so this script will exit.(B[m
  Creating diagnostics file...

[32mAll relevant logs have been anonymously uploaded to https://0x0.st/zO2k.zip
[32mCopy and paste [31m[logs.zip](https://0x0.st/zO2k.zip)[32m in the GitHub issue.(B[m
Make sure the suite is up-to-date before reporting an issue. It might've been fixed already.
Try running the build again at a later time.
[logs.zip](https://github.com/jb-alvarado/media-autobuild_suite/files/3470154/logs.zip)
```

[logs.zip](https://github.com/jb-alvarado/media-autobuild_suite)

**Comment by 1480c1** [comment]
``` none
... -LD:/media-autobuild_suite/local64/lib D:/media-autobuild_suite1/msys64/mingw64/x86_64-w64-mingw32/lib/libwinmm.a 
```

@wiiaboo / @Helenerineium , I just noticed the space between the `-L*` and the `...libwinmm`, Any idea where that space came from?

**Comment by wiiaboo** [comment]
The problem maybe that it's using absolute paths instead of the normal unix paths, and not using `-l` and instead the abspath to winmm.  I didn't notice this.

**Comment by jkchoi2** [comment]
I did not modify media-autobuild_suite.
And I am compiling in Windows 10 64-bit environment.

**Comment by wiiaboo** [comment]
I'm not saying you changed anything, just that openal may be generating a broken pkgconfig.

**Comment by wiiaboo** [comment]
Probably fixed with ab47fd3
